### PR TITLE
Refactoring endpoint

### PR DESCRIPTION
**Pull request by elenagensch**

### CODE REVIEW

File: packages/IMAPClient-Core.package/ICEndPoint.class/instance/parseListResponse.st [hunk 1]
```diff
@@ -1,4 +1,4 @@
-protocol handling
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICEndPoint.class/instance/parseListResponse.st]: please classify again

File: packages/IMAPClient-Core.package/ICEndPoint.class/instance/logout.st [hunk 2]
```diff
@@ -1,8 +1,5 @@
 login
 logout
 
-	self checkConnected.
-	self sendDirectCommand: ICCommand logout.
-	self close.
-
-	^ true
\ No newline at end of file
+	"Error shouldBeImplemented."
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICEndPoint.class/instance/logout.st]: I assume this was for debugging, but please remove before merging :)

File: packages/IMAPClient-Core.package/ICEndPoint.class/instance/login.st [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-	(self throwErrorIfNotLoggedIn) ifFalse: [^ false].
-  	(self throwErrorIfAccountInvalid) ifFalse: [^ false].
+	(self connectionHandler throwErrorIfNotLoggedIn) ifFalse: [^ false].
+  	(self connectionHandler throwErrorIfAccountInvalid) ifFalse: [^ false].
 	
-	self sendLoginCommand.
+	self connectionHandler sendLoginCommand.
 	
-	(self throwErrorIfNotLoggedIn) ifFalse: [^ false].
+	"(self throwErrorIfNotLoggedIn) ifFalse: [^ false]."
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICEndPoint.class/instance/login.st:9]: uncomment or remove

File: packages/IMAPClient-Core.package/ICEndPoint.class/instance/listFoldersIn.using..st [hunk 4]
```diff
@@ -1,10 +1,10 @@
-protocol handling - fetch
+storage
 listFoldersIn: root using: filter
 
-	self 
+	"self 
 		checkConnected;
-		checkLoggedIn.
+		checkLoggedIn."
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICEndPoint.class/instance/listFoldersIn.using..st]: uncomment or remove

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/class/defaultStreamClass.st [hunk 5]
```diff
@@ -0,0 +1,4 @@
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/class/defaultStreamClass.st]: please classify

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/class/errorInvalidConnectionConfiguration.st [hunk 6]
```diff
@@ -0,0 +1,4 @@
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/class/errorInvalidConnectionConfiguration.st]: please classify

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/class/refreshConnectionIntervalInSeconds.st [hunk 7]
```diff
@@ -0,0 +1,4 @@
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/class/refreshConnectionIntervalInSeconds.st]: please classify

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/checkConnected.st [hunk 8]
```diff
@@ -1,4 +1,4 @@
-error handling
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/checkConnected.st]: please classify again

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/checkForBadRequest.st [hunk 9]
```diff
@@ -1,11 +1,12 @@
-error handling
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/checkForBadRequest.st]: please classify again

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/checkForParameterError.st [hunk 10]
```diff
@@ -1,4 +1,4 @@
-error handling
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/checkForParameterError.st]: please classify again

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/checkLoggedIn.st [hunk 11]
```diff
@@ -1,4 +1,4 @@
-login
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/checkLoggedIn.st]: please classify again

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/deleteEmail..st [hunk 12]
```diff
@@ -0,0 +1,6 @@
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/deleteEmail..st]: please classify again

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/expungeEmails.st [hunk 13]
```diff
@@ -0,0 +1,4 @@
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/expungeEmails.st]: please classify

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/fetchFlagsForUIDs.to..st [hunk 14]
```diff
@@ -0,0 +1,4 @@
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/fetchFlagsForUIDs.to..st]: please classify

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/fetchHeadersForUIDsFrom.to..st [hunk 15]
```diff
@@ -0,0 +1,4 @@
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/fetchHeadersForUIDsFrom.to..st]: please classify

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/fetchNextResponse.st [hunk 16]
```diff
@@ -1,4 +1,4 @@
-protocol handling - fetch
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/fetchNextResponse.st]: please classify

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/getBodyFor..st [hunk 17]
```diff
@@ -0,0 +1,5 @@
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/getBodyFor..st]: please classify

File: packages/IMAPClient-Core.package/ICEndPoint.class/instance/currentUIDs.st [hunk 18]
```diff
@@ -1,11 +1,11 @@
...
-		removeFirst;
-		removeFirst; 
-		yourself)
\ No newline at end of file
+	"self isConnected 
+		ifTrue: ["^ (self lastResponse first subStrings: ' ') asOrderedCollection 
+			removeFirst;
+			removeFirst; 
+			yourself"]
+		ifFalse: [^ OrderedCollection new.]"
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICEndPoint.class/instance/currentUIDs.st]: remove commented out parts before merging

File: packages/IMAPClient-Core.package/ICEndPoint.class/class/newWith..st [hunk 19]
```diff
@@ -0,0 +1,9 @@
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICEndPoint.class/class/newWith..st]: please classify

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/throwErrorIfNotLoggedIn.st [hunk 20]
```diff
@@ -1,8 +1,8 @@
-login
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/throwErrorIfNotLoggedIn.st]: please classify

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/storedCommands.st [hunk 21]
```diff
@@ -0,0 +1,4 @@
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/storedCommands.st]: please classify

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/sendLoginCommand.st [hunk 22]
```diff
@@ -1,4 +1,4 @@
-login
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/sendLoginCommand.st]: please classify

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/sendCommand..st [hunk 23]
```diff
@@ -1,4 +1,4 @@
-connection
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/sendCommand..st]: please classify

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/select..st [hunk 24]
```diff
@@ -0,0 +1,4 @@
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/select..st]: please classify

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/resetFlag.of..st [hunk 25]
```diff
@@ -0,0 +1,4 @@
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/resetFlag.of..st]: please classify

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/moveEmail.to..st [hunk 26]
```diff
@@ -0,0 +1,4 @@
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/moveEmail.to..st]: please classify

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/listFoldersIn.using..st [hunk 27]
```diff
@@ -0,0 +1,10 @@
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/listFoldersIn.using..st]: please classify

File: packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/folderFromResponse..st [hunk 28]
```diff
@@ -0,0 +1,19 @@
+as yet unclassified
```
Comment (laugengebaeck) [inline packages/IMAPClient-Core.package/ICConnectionHandler.class/instance/folderFromResponse..st]: please classify

**Review by laugengebaeck — APPROVED** [review]
This looks very good! Nicely refactored :+1: 

Please don't forget to remove commented out lines and classify all methods before merging :)

**Comment by laugengebaeck** [comment]
Another idea: If you want to increase coverage a bit again, you might want to make the tests call the `ICEndpoint` methods (which in turn use the `ICConnectionHandler` methods) instead of calling the `ICConnectionHandler` methods directly :)

**Review by elenagensch — COMMENTED** [review]
Nice! We might check in the future whether there are still some unremoved methods in ICEndpoint.

**Pull request merged**